nir/src/main/scala/scala/scalanative/nir/serialization/BinaryDeserializer.scala

```diff
@@ -40,10 +41,18 @@ final class BinaryDeserializer(buffer: ByteBuffer) {
 
   private def getInts(): Seq[Int] = getSeq(getInt)
 
-  private def getStrings(): Seq[String] = getSeq(getString)
   private def getString(): String = {
```

Comment:
It seems that this method is now only used in one place, namely the `case T.StringVal` at the end of the file. I suggest inlining it there, so that it's obvious that its logic is only relevant for string literals, and nothing else.